src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,104 @@
+            directory.mkdir(); // Create if directory does not exist
+        }
+        // Check if duke.txt exist in the data directory
+        File file = new File("./data/duke.txt");
+        if (!file.exists()) {
+            file.createNewFile(); // Create if file does not exist
+        }
+        TaskList list = new TaskList();
+        // Read from duke.txt and store information back into task list
+        Scanner fc = new Scanner(file);
+        int index = 0;
+        while (fc.hasNext()) {
+            String[] str = fc.nextLine().split(",");
+            if (str[0].equals("T")) {
+                if (str[1].equals("1")) {
+                    Task newTask = new ToDo(str[2], true);
+                    list.addTask(newTask);
+                } else {
+                    Task newTask = new ToDo(str[2], false);
+                    list.addTask(newTask);
+                }
+                index++;
+            } else if (str[0].equals("D")) {
+                try {
+                    if (str[1].equals("1")) {
+                        LocalDate date = LocalDate.parse(str[3]);
+                        Task newTask = new Deadline(str[2], date, true);
+                        list.addTask(newTask);
+                    } else {
+                        LocalDate date = LocalDate.parse(str[3]);
+                        Task newTask = new Deadline(str[2], date, false);
+                        list.addTask(newTask);
+                    }
+                    index++;
+                } catch (DateTimeParseException ex) {
+                    throw new DukeException("INVALID DATE FORMAT FOUND IN DUKE.TXT FILE. UNABLE TO READ FROM FILE.");
+                }
+            } else if (str[0].equals("E")) {
+                try {
+                    if (str[1].equals("1")) {
+                        LocalDate date = LocalDate.parse(str[3]);
+                        Task newTask = new Event(str[2], date, true);
+                        list.addTask(newTask);
+                    } else {
+                        LocalDate date = LocalDate.parse(str[3]);
+                        Task newTask = new Event(str[2], date, false);
+                        list.addTask(newTask);
+                    }
+                    index++;
+                } catch (DateTimeParseException ex) {
+                    throw new DukeException("INVALID DATE FORMAT FOUND IN DUKE.TXT FILE. UNABLE TO READ FROM FILE.");
+                }
+            } else {
+                throw new DukeException("INVALID TASK FOUND IN DUKE.TXT FILE. UNABLE TO READ FROM FILE.");
+            }
+        }
+        return list;
+    }
+
+    /**
+     * Writes data from the arraylist into file.
+     *
+     * @param list an arraylist containing task objects.
+     * @throws IOException if there's error arising from I/O methods.
+     */
+    public static void writeFile(TaskList list) throws IOException {
+        FileWriter fw = new FileWriter("./data/duke.txt");
```

Comment:
Should create the directory and the file, if they didn't exist to avoid `FileNotFoundException`.